### PR TITLE
Rust const-fn implementation (solution 6)

### PR DESCRIPTION
**Pull request by Sycration**
## Description

Rust has a const function feature that lets you run code at compile time. I ported the original C++ version to rust and used this feature. This makes it unfaithful as the size of the sieve is determined at compile time. 

## Contributing requirements

* [x] I read the contribution guidelines in CONTRIBUTING.md.
* [x] I placed my solution in the correct solution folder.
* [x] I added a README.md with the right badge(s).
* [x] I added a Dockerfile that builds and runs my solution.
* [x] I selected `drag-race` as the target branch.
* [x] All code herein is licensed compatible with BSD-3.

### CODE REVIEW

**Comment by rbergen** [comment]
@Sycration Please indicate when we can consider this PR to be stable.

**Comment by Sycration** [comment]
@rbergen  I just cleaned up the multithreaded version and it should be all stable & good now.

**Review by rbergen — APPROVED** [review]
LGTM

File: PrimeRust/solution_6/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM rustlang/rust:nightly as build
```
Comment (rbergen) [inline PrimeRust/solution_6/Dockerfile]: Please pin the base docker image to a specific version. We've recently had some spontaneous breakages in our CI due to unpinned version shifts.

Comment (Sycration) [reply PrimeRust/solution_6/Dockerfile]: fixed, with specific digest hash

Comment (rbergen) [reply PrimeRust/solution_6/Dockerfile]: Thank you!

**Review by marghidanu — APPROVED** [review]

**Comment by marghidanu** [comment]
Let's wait for the CI to run before merging.

**Pull request merged**